### PR TITLE
add alert for recaptcha

### PR DESCRIPTION
**Pull request by mh4ck**
Nothing was thrown if the reCAPTCHA form is shown to the server and the process quit silently.

### CODE REVIEW

**Comment by TimeForANinja** [comment]
🤔 do i need to understand why you switched to request?

File: lib/info.js [hunk 1]
```diff
@@ -54,6 +55,11 @@ exports.getBasicInfo = (id, options, callback) => {
       }
     }
 
+    const captchaForm = util.between(body, '<div class="g-recaptcha"', '>');
+    if(captchaForm) {
```
Comment (fent) [inline lib/info.js]: if using a variable right away and only once, just put its value into where it's used

File: lib/info.js [hunk 2]
```diff
@@ -54,6 +55,11 @@ exports.getBasicInfo = (id, options, callback) => {
...
+      return callback(Error("reCAPTCHA form shown"));
```
Comment (fent) [inline lib/info.js]: use single quotes, and let's reword this to have a little more information on why the form was shown. is it shown when made too many requests, such as the 429 error? does the form say anything?

**Comment by mh4ck** [comment]
> interested why you switched to request 🤔

Shit it shouldn't be in this Pull Request - also strange i'm pretty sure i made the pull request before i changed to request.

But to answer your question. We are not able to send proxy requests to YouTube with the current miniget implementation. With request it's working well.

File: lib/info.js (hunk 2 above)
Comment (mh4ck) [reply lib/info.js]: That's the HTML code of the form:

```
<div id="content" class="  content-alignment" role="main"><p class='largeText'>Sorry for the interruption. We have been receiving a large volume of requests from your network.</p>

<p>To continue with your YouTube experience, please fill out the form below.</p>
<form action="/das_captcha" method="POST">
<div class="g-recaptcha" data-sitekey="6Lf39AMTAAAAALPbLZdcrWDa8Ygmgk_fmGmrlRog"></div>
<br/>
<input type="hidden" name="next" value="/watch?v=s9xtPSweJNI&amp;hl=en&amp;bpctr=1574634970">
<input type="hidden" name="session_token" value="QUFFLUhqbE9talladHA2NmxZUk5CenQxZXU3WlQwSkYtUXxBQ3Jtc0ttNjB0UFgzRzU1bXRvU054VnJ2X01sMGh4Y3RKZHR0VUwtNldWZ045ZS1EbWFXbHJ1Rkh2TXYxTDJoNDljenFtOGtVbjlwZ3pTdktEZ3VkdjBFMnBBTEtGX0VqMUVUdzJ0NG13UV9ILUtzQzdvcW9IdTd6VFZTUzFfcjhJTWc2RF9BZUhXM3JSRVJENmJVMnBTSDVmaHBTUUJ6T1E=">
<input type="hidden" name="action_recaptcha_verify2" value="1">
<input type="submit" value="Submit">
</form>
```

File: lib/info.js (hunk 1 above)
Comment (mh4ck) [reply lib/info.js]: Isn't it more readable to use it like this. 
In my opinion there shouldn't be to much text in an if.

Also it's easy to debug by adding a simple console.log between this lines. 

I read it in a style guide but have forgotten which one, i think it's very usefull.

**Comment by mh4ck** [comment]
> hi, thank you for taking the time to make the PR. besides the comments, I have two more things to address
> 
> * do you mind adding a test? look into the entries in `test/files/refresh.js`, although we can't easily replicate and get this error, we can add a `transform` in that file, adding in the error ourselves
> * I know you prefer request, but that's a separate change, and should be in it own PR if at all. we can have a discussion there about it

Sure i will try to write a test. :)
And also you are right, the change to request should be in another PR, what is the best practise to remove it from this PR?

**Comment by fent** [comment]
> And also you are right, the change to request should be in another PR, what is the best practise to remove it from this PR?

`git rebase --interfactive` https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

then after you drop the commit(s), you can force-push your changes to this branch

File: lib/info.js (hunk 2 above)
Comment (fent) [reply lib/info.js]: how about the text from the `largeText` element?

File: lib/info.js (hunk 1 above)
Comment (fent) [reply lib/info.js]: hmm that's a good point

Comment (TimeForANinja) [reply lib/info.js]: The problem with javascript is that since it's designed to be send over the web and interpreted live you (in general) want to reduce it as much as possible.
the performance should be about the same
removing the variable will end up with a line still easy to read and under 80 chars

File: lib/info.js (hunk 2 above)
Comment (TimeForANinja) [reply lib/info.js]: would be cool to export `data-sitekey` and `session_token` since (as far as i know) those two are everything that's needed to solve the captcha
maybe an optional `option.captchaSolver = (siteKey, sessionToken) => { ... }` ?

**Comment by fent** [comment]
revert or remove the unwanted commits, instead of manually editing the files yourself and committing again.

use [`git rebase --interactive`](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) to remove, or `git revert` to revert
